core/vibe/core/drivers/libasync.d

```diff
@@ -1315,6 +1329,7 @@ final class LibasyncTCPConnection : TCPConnection/*, Buffered*/ {
 	}
 
 	void acquireReader() {
+		assert(!m_settings.writer.isWaiting, "Acquiring reader that is already in use.");
```

Comment:
I was about to refactor those methods into the `Waiter` struct... and then went the copy/past route ;)